propolis/src/hw/nvme/ns.rs

```diff
@@ -0,0 +1,226 @@
+use std::collections::VecDeque;
+use std::sync::{Arc, Mutex};
+
+use crate::block::*;
+use crate::hw::nvme::bits::RawCompletion;
+use crate::hw::nvme::cmds::{self, NvmCmd};
+use crate::{common, dispatch::DispCtx};
+
+use super::bits::{self, RawSubmission};
+use super::cmds::{Completion, ReadCmd, WriteCmd};
+use super::queue::{CompQueue, SubQueue};
+use super::NvmeError;
+
+const BLOCK_SZ: u64 = 512;
```

Comment:
Presumably we're going to want a TODO here to make this something that's less constant or at least not guaranteed to be 512 bytes. I imagine that we're going to want to advertise at least 4k sectors over time.

Reply:
Yea, I just hadn't gotten around to that because Format isn't supported yet anyways. I'll add a note for now.